File: Creational/Builder/PHP/products/EngineV8.php
<?php

require_once(__DIR__.'/../interfaces/IEngine.php');

/**
 * Concrete component.
 */

class EngineV8 implements IEngine
{
	public function __construct($volume = 4.0)
	{
		$this->_volume = $volume;
	}

	public function getType()
	{
		return 'V8';
	}

	public function getVolume()
	{
		return $this->_volume;
	}

	private $_volume;
}

File: Creational/Builder/Java/products/EngineV8.java
package patterns.creational.builder.products;

import patterns.creational.builder.interfaces.IEngine;

public class EngineV8 implements IEngine {
    /**
     * @param volume
     */
    public EngineV8(final float volume) {
        _volume = volume;
    }

    /**
     * @return String
     */
    public String getType() {
        return "V8";
    }

    /**
     * @return float
     */
    public float getVolume() {
        return _volume;
    }

    final private float _volume;
}
